scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -556,6 +557,32 @@ object Execution {
     final case class Force[T](pipe: TypedPipe[T]) extends ToWrite
     final case class ToIterable[T](pipe: TypedPipe[T]) extends ToWrite
     final case class SimpleWrite[T](pipe: TypedPipe[T], sink: TypedSink[T]) extends ToWrite
+
+    /**
+     * Optimize these writes into new writes and provide a mapping from
+     * the original TypedPipe to the new TypedPipe
+     */
+    def optimizeWriteBatch(writes: List[ToWrite], rules: Seq[Rule[TypedPipe]]): HMap[TypedPipe, TypedPipe] = {
+      val dag = Dag.empty(typed.OptimizationRules.toLiteral)
+      val (d1, ws) = writes.foldLeft((dag, List.empty[Id[_]])) {
+        case ((dag, ws), Force(p)) =>
```

Comment:
is this force a Force from a force to disk or a force to disk execution?

Reply:
ForceToDiskExecution.